### PR TITLE
Blood deficiency quirk now increases oxygen damage for bloodless species

### PR DESCRIPTION
**Pull request by Mickyan**
## About The Pull Request

Makes it so the blood deficiency doubles the oxygen damage you take instead if it's selected by a species with no blood

## Why It's Good For The Game
This quirk does nothing if your species does not have blood so people have been exploiting it for free points. Since blood loss typically causes oxygen damage this seems like an appropriate alternative abstraction for someone with compromised oxygenation.

## Changelog

:cl:
balance: Bloodless species with the blood deficiency quirk now take increased oxygen damage
/:cl:

### CODE REVIEW

File: code/datums/quirks/negative.dm [hunk 1]
```diff
@@ -60,17 +60,32 @@
 	medical_record_text = "Patient requires regular treatment for blood loss due to low production of blood."
 	hardcore_value = 8
 	processing_quirk = TRUE
+	var/bloodless_species = FALSE
+
+/datum/quirk/blooddeficiency/add()
```
Comment (JohnFulpWillard) [inline code/datums/quirks/negative.dm:65]: ``add()`` is called before they check for ``processing_quirk``, I don't think you have to stop processing here at all.

**Comment by tralezab** [comment]
a quirk about blood deficiency doesn't make any sense on a species that has no blood, and the solution isn't adding extra oxygen damage.

**Comment by RikuTheKiller** [comment]
Not the solution, prevent bloodless from even taking it. However, this seems like a great quirk idea on it's own.

**Comment by Mothblocks** [comment]
Not that I disagree with blocking this PR but @tralezab you and I have talked about species blocking quirks to be nontrivial with some spooky edge cases. Was that a PR you were going to make?

**Comment by Floofies** [comment]
FYI! @RikuTheKiller I am working on a system to block species/quirk selections in TGUI, I will PR it soon! Multiple downstreams of TG seem to want the feature as well and for the same reasons. The blacklist format looks like this:
```// A list of species and quirks which can not be used with each other.
var/static/list/quirk_species_blacklist = list(
  SPECIES_PLASMAMAN = list("Blood Deficiency"),
  SPECIES_SKELETON = list("Blood Deficiency")
)```

**Comment by Mothblocks** [comment]
@Floofies 

There are 3 requirements I have for it

1. Switching species must rebalance your quirks.
2. Before switching species, a UI confirm prompt must come up showing you what quirks are going to be changed.
3. It must work with quirk filtering, so we can add it to stuff later on and have it apply.

**Pull request closed without merging**